src/coreclr/gc/gc.cpp

```diff
@@ -25193,7 +25226,8 @@ int gc_heap::calculate_new_heap_count ()
     float throughput_cost_percents[dynamic_heap_count_data_t::sample_size];
     for (int i = 0; i < dynamic_heap_count_data_t::sample_size; i++)
     {
-        dynamic_heap_count_data_t::sample& sample = dynamic_heap_count_data.samples[i];
+        int sample_index = (dynamic_heap_count_data.sample_index + i) % dynamic_heap_count_data_t::sample_size;
+        dynamic_heap_count_data_t::sample& sample = dynamic_heap_count_data.samples[sample_index];
```

Comment:
I added this a while ago because I wanted to ensure that the tcps and dprintfs were in chronological order, but is this necessary?

Reply:
since none of our eval has anything to do with order of the samples this does not seem necessary.

Reply:
good point - it was an early experiment (wanting to look at the startup patterns) but isn't relevant now